### PR TITLE
chore: update temporary entries for cloud.google.com/go/spanner in go.mod to public versions supporting PG feature

### PR DESCRIPTION
**Pull request by rahul2393**
Fixes #<issue_number_goes_here>

- [x] Tests pass
- [x] Appropriate changes to README are included in PR

### What is the motivation behind this?

We need this because of a 'dependency hell' situation:

1. `grpc-go` makes breaking changes between minor releases of their module.
2. tidb v1.1.0-beta [depends on a `grpc/naming` API which was removed in v1.30](https://github.com/etcd-io/etcd/issues/12124). We depend on etcd v0.5.0-alpha indirectly via tidb v1.1.0-beta.

This situation would be solved by bumping tidb to version which depends on etcd v3.6.0+, which does not make use of the removed grpc-go API.

This module is used to solve this dependency hell, by adding the `google.golang.org/grpc/naming` package into version v1.44 of grpc-go.

It is the smallest patch that can be applied to solve this particular 'dependency hell' issue with grpc-go. It will not fix other issues when packages depend on other, newer packages.

## When can it be removed?

This must be removed when tidb does not need grpc/naming.

### CODE REVIEW

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM

File: naming/dns_resolver.go [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package naming
```
Comment (bharadwaj-aditya) [inline naming/dns_resolver.go:19]: Please add a comment linking where this snippet has been taken from

File: naming/naming.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package naming defines the naming API and related data structures for gRPC.
+//
+// This package is deprecated: please use package resolver instead.
```
Comment (bharadwaj-aditya) [inline naming/naming.go:21]: please add a comment as to where this snippet has been taken from

File: naming/dns_resolver.go (hunk 1 above)
Comment (rahul2393) [reply naming/dns_resolver.go:19]: I added in naming/README https://github.com/cloudspannerecosystem/harbourbridge/pull/265/files#diff-e48c26b9bc60a5770dd43c73b76b599d9672681ffef4f90be8b20137c3593514R4

Comment (rahul2393) [reply naming/dns_resolver.go:19]: Do you think we need to add here too?

**Comment by Deep1998** [comment]
LGTM, thanks for this fix. Please update the PR description as well though. You can simply copy paste the readme contents.

File: naming/dns_resolver.go (hunk 1 above)
Comment (bharadwaj-aditya) [reply naming/dns_resolver.go:19]: Makes sense to add a comment here too, just to make it clear that this is going to be a copy of the source and not to add any custom code.

Comment (rahul2393) [reply naming/dns_resolver.go:23]: Done!! Github is not showing `Outdated` for some reasons.

File: naming/naming.go (hunk 2 above)
Comment (rahul2393) [reply naming/naming.go:25]: Done!! Github is not showing `Outdated` for some reasons.

**Pull request merged**